python/lsst/pipe/tasks/imageDifference.py

```diff
@@ -499,13 +606,19 @@ def runDataRef(self, sensorRef, templateIdList=None):
 
                     self.log.info("Selected %d / %d sources for Psf matching (%d for control sample)"
                                   % (len(kernelSources), len(selectSources), len(controlSources)))
+                else:
+                    # AL subtraction will do its own source detection
+                    kernelSources = None
+
                 allresids = {}
                 if self.config.doUseRegister:
```

Comment:
This may also be beyond the scope of this ticket, but I would like to see everything in this `if self.config.doUseRegister:` code block moved to its own method.